docs/core-features/table_lookups_basics.md

```diff
@@ -0,0 +1,186 @@
+circuit = f.compile(inputset)
+
+sample = [
+    [0, 1],
+    [2, 3],
+    [3, 0],
+]
+expected_output = [
+    [0, 1],
+    [4, 27],
+    [9, 0]
+]
+actual_output = circuit.encrypt_run_decrypt(np.array(sample))
+
+assert np.array_equal(actual_output, expected_output)
+```
+
+### TLU on the most significant bits
+
+For lot of use-cases, like for example in Machine Learning, it is possible to replace the table lookup `y = T[i]` by some `y = T'[i']`, where `i'` only has the most significant bits of `i` and `T'` is a much shorter table, and still maintain a good accuracy of the function. The interest of such a method stands in the fact that, since the table `T'` is much smaller, the corresponding TLU will be done much more quickly.
+
+There are different flavors of doing this in Concrete. We describe them quickly here, and refer the user to the [poweruser documentation](../dev/compilation/compiler_internals.md) for more explanations.
+
+The first possibility is to set `i'` as the truncation of `i`: here, we just take the most significant bits of `i`. This is done with `fhe.truncate_bit_pattern`.
+
+```python
+from concrete import fhe
+import numpy as np
+
+table = fhe.LookupTable([i**2 for i in range(16)])
+lsbs_to_remove = 1
+
+
+@fhe.compiler({"x": "encrypted"})
+def f(x):
+    return table[fhe.truncate_bit_pattern(x, lsbs_to_remove)]
+
+
+inputset = range(16)
+circuit = f.compile(inputset)
+
+for i in range(16):
+    rounded_i = int(i / 2**lsbs_to_remove) * 2**lsbs_to_remove
+
+    assert circuit.encrypt_run_decrypt(i) == rounded_i**2
+```
+
+The second possibility is to set `i'` as the rounding of `i`: here, we take the most significant bits of `i`, and increment by 1 if ever the most significant ignored bit is a 1, to round.  This is done with `fhe.round_bit_pattern`. It's however a bit more complicated, since rounding may make us go "out" of the original table. Remark how we enlarged the original table by 1 index.
+
+```python
+from concrete import fhe
+import numpy as np
+
+table = fhe.LookupTable([i**2 for i in range(17)])
+lsbs_to_remove = 1
+
+
+def our_round(x):
+    float_part = x - np.floor(x)
+    if float_part < 0.5:
+        return int(np.floor(x))
+    return int(np.ceil(x))
+
+
+@fhe.compiler({"x": "encrypted"})
+def f(x):
+    return table[fhe.round_bit_pattern(x, lsbs_to_remove)]
+
+
+inputset = range(16)
+circuit = f.compile(inputset)
+
+for i in range(16):
+    rounded_i = our_round(i * 1.0 / 2**lsbs_to_remove) * 2**lsbs_to_remove
+
+    assert (
+        circuit.encrypt_run_decrypt(i) == rounded_i**2
+    ), f"Miscomputation {i=} {circuit.encrypt_run_decrypt(i)} {rounded_i**2}"
+```
+
+Finally, for `fhe.round_bit_pattern` (and soon for `fhe.truncate_bit_pattern`), there exist an `exactness=fhe.Exactness.APPROXIMATE` option, which can make computations even faster, at the price of having a few (minor) differences between cleartext computations and encrypted computations.
```

Comment:
I think we should never commit for the future in the documentation. As previous comment we don't say reducing input bitwidth speedup the computation but here you said its even faster

Reply:
if I replace "soon" by "later", does that work for you @BourgerieQuentin ? I just would like to tell them it's coming, because, as a user, it's the option I find the most natural to use and have

Reply:
As you want, but for me in the documentation we should document only what we have right now nothing else.
Maybe @yuxizama have an opinion on that?

Reply:
ok I remove it, but as a user, I find it strange and disappointing the option doesn't exist